api/offer_test.go

```diff
@@ -161,6 +207,280 @@ var _ = Describe("Offer Template Handler", func() {
 			Expect(obj["description"]).To(Equal("Contents: [34 34] does not validate as RequiredJSONObject;;"))
 		})
 
+		Describe("Invalid filters format", func() {
+			It("should return status code 422 if invalid filter format", func() {
+				name := "New Awesome Game"
+				productID := "com.tfg.example"
+				gameID := "game-id"
+				contents := `{"gems": 5, "gold": 100}`
+				period := `{"max": 1}`
+				frequency := `{"every": "24h"}`
+				trigger := `{"from": 1487280506875, "to": 1487366964730}`
+				filters := `{"level": {"eq": 5}}`
```

Comment:
They need to be strings just because of the way the filter comparison is made, for equality I always assume the value is a string, supporting integers would add some extra complexity to the SQL query.